### PR TITLE
bug fix for 1 plus 5 minting

### PR DESCRIPTION
**Pull request by Maar-io**

### CODE REVIEW

File: contract/contracts/AstarDegens.sol [hunk 1]
```diff
@@ -1248,13 +1248,13 @@ contract AstarDegens is ERC721Enumerable, Ownable {
   function mint(uint256 _mintAmount) public payable {
     uint256 supply = totalSupply();
     require(!paused, "Degens are lazy, call the king to wake them up");
-    require(_mintAmount > 0);
+    require(_mintAmount > 0, "Mint amount is 0");
     require(_mintAmount <= maxMintAmount, "Degen tribe is max 5 apes");
```
Comment (Dinonard) [inline contract/contracts/AstarDegens.sol:1252]: You should remove this two checks and just replace them with what you have on line 1257.

**Review by Dinonard — COMMENTED** [review]
Once you add `await` to your tests, they will start failing.
It seems they were incorrectly passing for you locally.

File: contract/contracts/AstarDegens.sol (hunk 1 above)
Comment (Maar-io) [reply contract/contracts/AstarDegens.sol:1252]: let's discuss. I do not agree

File: contract/test/astarDegens-test.js [hunk 2]
```diff
@@ -132,30 +156,50 @@ describe("AstarDegens contract", function () {
...
+        await ad.connect(bob).mint(1, {
+          value: degenCost.mul(1),
+        })
+      )
+        .to.emit(ad, "Transfer")
+        .withArgs(ethers.constants.AddressZero, bob.address, tokenId.add('1'));
+      expect(await ad.totalSupply()).to.equal(1);
+
+      // should fail to mint additional one in new mint call
+      expect(ad.connect(bob).mint(5, { value: degenCost }))
```
[the remark this answers is not in the document]
Comment (Maar-io) [reply contract/test/astarDegens-test.js]: done

File: contract/test/astarDegens-test.js [hunk 3]
```diff
@@ -132,30 +156,50 @@ describe("AstarDegens contract", function () {
 
       // should fail to mint additional one in new mint call
       expect(ad.connect(bob).mint(1, { value: degenCost }))
```
[the remark this answers is not in the document]
Comment (Maar-io) [reply contract/test/astarDegens-test.js]: yes

**Pull request merged**